dataset.py

```diff
@@ -0,0 +1,42 @@
+import json
+import random
+
+dataset_seed = 1337
+class Dataset:
+    def __init__(self, file, max_queries, max_input_tokens, max_output_tokens):
+        self.dataset_list = [query for query in read_jsonlines(file, max_queries=max_queries, max_input_tokens=max_input_tokens, max_output_tokens=max_output_tokens)]
+        random.Random(dataset_seed).shuffle(self.dataset_list)
+        self.index = 0
+
+    def get_next_n_queries(self, n):
+        max_index=len(self.dataset_list)
+        next_n_indices = [i % max_index for i in range(self.index, self.index+n)] 
+        self.index = (self.index + n) % max_index
+        return [self.dataset_list[i] for i in next_n_indices]
+
+
+def read_jsonlines(filename, max_queries=1000, max_input_tokens=1024, max_output_tokens=1024):
+    with open(filename, 'r', encoding='utf-8') as file:
+        next(file) #First line for metadata
```

Comment:
I find this line confusing,
I guess it's correct as it works for you (=skip the first line), but the reading isn't straightforward

I usually prefer something more explicit as:
```
first_line = True
for line in files:
  if first_line: 
    first_line = False
    continue
```
or 
```
for idx, line in enumerate(files):
  if idx == 0:
      # skip the first line, it contains metadata
      continue
```